src/common/style/icon_color_css.ts

```diff
@@ -1,5 +1,23 @@
 import { css } from "lit-element";
 
+export interface StateColor {
+  state: string;
+  color?: string;
+}
+
+export const computeStateColor = (
+  state: string,
+  stateColors: StateColor[]
+): string => {
+  for (const stateColor of stateColors) {
+    if (stateColor.state === state) {
+      return stateColor.color || "--paper-item-icon-active-color";
+    }
+  }
+
+  return "";
```

Comment:
should we return undefined here instead? Also, this is not returning the true state color. Maybe the function should be called computeCustomStateColor? or something similar